test/functional/output_kafka.go

```diff
@@ -0,0 +1,153 @@
+		WithCmdArgsToInitContainer([]string{"/bin/bash", "/etc/kafka-configmap/init.sh"}, 1).
+		AddVolumeMountToInitContainer("brokerconfig", "/etc/kafka-configmap", "", false, 1).
+		AddVolumeMountToInitContainer("configkafka", "/etc/kafka", "", false, 1).
+		AddVolumeMountToInitContainer("extensions", "/opt/kafka/libs/extensions", "", false, 1)
+
+	cmdCreateTopicAndDeployBroker := "./bin/kafka-server-start.sh /etc/kafka-configmap/server.properties"
+
+	log.V(2).Info("Adding container", "name", name)
+	b.AddContainer(kafkaBrokerContainerName, ImageRemoteKafka).
+		AddEnvVar("CLASSPATH", "/opt/kafka/libs/extensions/*").
+		AddEnvVar("KAFKA_LOG4J_OPTS", "-Dlog4j.configuration=file:/etc/kafka/log4j.properties").
+		AddEnvVar("JMX_PORT", strconv.Itoa(int(kafkaJMXPort))).
+		AddContainerPort("inside", kafkaInsidePort).
+		AddContainerPort("outside", kafkaOutsidePort).
+		AddContainerPort("jmx", kafkaJMXPort).
+		//WithCmd("./bin/kafka-server-start.sh /etc/kafka-configmap/server.properties").
+		WithCmdStringSlice([]string{"/bin/bash", "-c", cmdCreateTopicAndDeployBroker}).
+		AddVolumeMount("brokerconfig", "/etc/kafka-configmap", "", false).
+		AddVolumeMount("configkafka", "/etc/kafka", "", false).
+		AddVolumeMount("brokerlogs", "/opt/kafka/logs", "", false).
+		AddVolumeMount("brokercert", "/etc/kafka-certs", "", false).
+		AddEnvVar("extensions", "/opt/kafka/libs/extensions").
+		AddEnvVar("data", "/var/lib/kafka/data").
+		End().
+		AddConfigMapVolume("brokerconfig", kafka.DeploymentName).
+		AddSecretVolume("brokercerts", kafka.DeploymentName).
+		AddEmptyDirVolume("brokerlogs").
+		AddEmptyDirVolume("extensions")
+
+	/////////////////////////////////////////////
+	//step c
+	topics := []string{kafka.AppLogsTopic}
+
+	for _, topic := range topics {
+		containername := kafka.ConsumerNameForTopic(topic)
+		// Deploy consumer
+		//consumerapp := kafka.NewKafkaConsumerDeployment(OpenshiftLoggingNS, topic)
+
+		log.V(2).Info("Creating Broker Consumer app")
+		///create topic using --bootstrap-server localhost:9093
+		cmdCreateTopic := fmt.Sprintf(`./bin/kafka-topics.sh --zookeeper localhost:2181 --create --if-not-exists --topic %s --partitions 1 --replication-factor 1 ;`, topic)
+		cmdRunConsumer := fmt.Sprintf(`./bin/kafka-console-consumer.sh --bootstrap-server %s --topic %s --from-beginning > /shared/consumed.logs ;`, "localhost:9092", topic)
+		cmdSlice := []string{"sleep 120;", cmdCreateTopic, cmdRunConsumer}
+		cmdJ := strings.Join(cmdSlice, "")
+		cmdCreateTopicAndDeployConsumer := []string{"/bin/bash", "-c", cmdJ}
+
+		b.AddContainer(containername, ImageRemoteKafka).
+			WithCmdStringSlice(cmdCreateTopicAndDeployConsumer).
+			AddVolumeMount("brokerconfig", "/etc/kafka-configmap", "", false).
+			AddVolumeMount("brokercert", "/etc/kafka-certs", "", false).
+			AddVolumeMount("shared", "/shared", "", false).
+			End().
+			AddSecretVolume("brokercert", kafka.DeploymentName).
+			AddEmptyDirVolume("shared")
+
+	}
+
+	return nil
```

Comment:
I propose to break this function into 3 smaller functions and not have one big function for all steps.